### PR TITLE
Add TG workflow and pipeline tests

### PR DESCRIPTION
**Pull request by tapspatel**

### CODE REVIEW

**Comment by tapspatel** [comment]
passing pipeline: https://github.com/tenstorrent/tt-metal/actions/runs/9022561794/job/24792364337?pr=8312

File: .github/workflows/tg-unit-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    secrets: inherit
+  TG-tests:
+    strategy:
+      # Do not fail-fast because we need to ensure all tests go to completion
+      # so we try not to get hanging machines
+      fail-fast: false
+      matrix:
+        runner-info: [
+          # TG
+          {arch: wormhole_b0, runs-on: ["config-tg"]},
```
Comment (tt-rkim) [inline .github/workflows/tg-unit-tests.yaml:20]: Have we thought about `in-service` or `out-of-service`? Or should we just have `in-service` and the absence of it implies `out-of-service` - per your infra cleanup proposal for labels>

Comment (tapspatel) [reply .github/workflows/tg-unit-tests.yaml:20]: We should have in-service and out-service labels for sure, adding

Comment (tapspatel) [reply .github/workflows/tg-unit-tests.yaml:20]: Ill starting adding it to all galaxy machines and t3ks

Comment (tt-rkim) [reply .github/workflows/tg-unit-tests.yaml]: 👍

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**